builder/triton/ssh.go

```diff
@@ -0,0 +1,36 @@
+package triton
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/multistep"
+	"golang.org/x/crypto/ssh"
+)
+
+func commHost(state multistep.StateBag) (string, error) {
+	driver := state.Get("driver").(Driver)
+	machineID := state.Get("machine").(string)
+
+	machine, err := driver.GetMachine(machineID)
+	if err != nil {
+		return "", err
+	}
+
+	return machine, nil
+}
+
+func sshConfig(state multistep.StateBag) (*ssh.ClientConfig, error) {
+	config := state.Get("config").(Config)
+
+	signer, err := ssh.ParsePrivateKey([]byte(config.Comm.SSHPrivateKey))
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up SSH config: %s", err)
+	}
+
+	return &ssh.ClientConfig{
+		User: config.Comm.SSHUsername,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+	}, nil
```

Comment:
Thanks for explaining.

> there is a special Triton SSH plugin in OpenSSH

Do you have any refs on this, I'd like to understand this a bit more.

> Also the Triton Go Cloud API Library provided by Joyent can currently only deal with private keys in files.

We should add some validation in `Prepare` and validate a few things, such as:
1) Communicator is `ssh` or `none`.
2) `ssh_password==""`
3) `ssh_private_key_file` is a file and exists unless `communicator==none`.